internal/controllers/machinedeployment/machinedeployment_sync.go

```diff
@@ -532,16 +532,20 @@ func (r *Reconciler) updateMachineDeployment(ctx context.Context, d *clusterv1.M
 
 // We have this as standalone variant to be able to use it from the tests.
 func updateMachineDeployment(ctx context.Context, c client.Client, d *clusterv1.MachineDeployment, modify func(*clusterv1.MachineDeployment)) error {
```

Comment:
fyi. We are using this only to set the revision annotation in MD
We never read this annotation from a MD in our entire code base.

So it shouldn't matter that we are losing the side-effect that after this call the passed in deployment already has the annotation (with it's new value)